core/src/main/java/org/jsmart/zerocode/core/runner/ZeroCodeMultiStepsScenarioRunnerImpl.java

```diff
@@ -161,21 +161,33 @@ private boolean executeSteps(RunNotifier notifier,
 
             correlLogger = ZerocodeCorrelationshipLogger.newInstance(LOGGER);
 
-            thisStep = extFileProcessor.resolveExtJsonFile(thisStep);
-            thisStep = extFileProcessor.createFromStepFile(thisStep, thisStep.getId());
+            Boolean wasExecSuccess = executeRetryWithSteps(notifier, description, scenarioExecutionState, scenario, thisStep);
+            if (wasExecSuccess != null) return wasExecSuccess;
+        }
+
+        return false;
+    }
 
-            Boolean wasExecSuccess = executeRetry(notifier,
+    private Boolean executeRetryWithSteps(RunNotifier notifier,
+                                          Description description,
+                                          ScenarioExecutionState scenarioExecutionState,
+                                          ScenarioSpec scenario, Step thisStep) {
+        thisStep = extFileProcessor.resolveExtJsonFile(thisStep);
+        List<Step> thisSteps = extFileProcessor.createFromStepFile(thisStep, thisStep.getId());
+        if(null == thisSteps || thisSteps.isEmpty()) thisSteps.add(thisStep);
+        Boolean wasExecSuccess = null;
+        for(Step step : thisSteps) {
+             wasExecSuccess = executeRetry(notifier,
                     description,
                     scenarioExecutionState,
                     scenario,
-                    thisStep);
-
+                    step);
             if (wasExecSuccess != null) {
                 return wasExecSuccess;
             }
         }
-
-        return false;
+        ;
```

Comment:
Can you remote this `;` which seems like a typo.
This shd be the final one I promise, then we will merge it!

Reply:
Can u fix this @anishjha93 ?
> You replied with 'done', but it is still there.